Update python_versions.json to rotate with 3.12

jpfeuffer opened the pull request:
## Description

<!-- Please include a summary of the change and which issue is fixed here. -->

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

jpfeuffer committed:
Update python_versions.json to rotate with 3.12

jpfeuffer commented:
There are some requests to still include 3.9 until 2025.
But I don't know where this ends. What if python releases 10 more versions until 25?
EDIT: According to https://www.python.org/downloads/ there are usually 5 active releases with 2 in security support mode only.

timosachsenberg commented:
do we use anywhere 3.9 ? e.g., conda, docs, binder, streamlit apps etc.?
check https://github.com/search?q=org%3AOpenMS%203.9&type=code

jpfeuffer committed:
too lazy to check all possible places. just build everything.

we really should make sure that all things that use pyopenms, follow our rules. we can't just continue building everything, just in case someone uses it somewhere.

jpfeuffer commented:
I don't know. I just build everything now.  I am not sure if a fifth one will work in the future, also when thinking about runtimes on bioconda etc.

poshul committed:
Merge branch 'develop' into jpfeuffer-patch-5

timosachsenberg committed:
Merge branch 'develop' into jpfeuffer-patch-5

timosachsenberg committed:
Merge branch 'develop' into jpfeuffer-patch-5

timosachsenberg commented:
seems changes are already included. will close
